build.gradle

```diff
@@ -167,6 +167,7 @@ project(":samza-core_$scalaVersion") {
     compile "org.scala-lang:scala-library:$scalaLibVersion"
     compile "org.slf4j:slf4j-api:$slf4jVersion"
     testCompile project(":samza-api").sourceSets.test.output
+    testCompile project(":samza-kv-rocksdb_$scalaVersion")
```

Comment:
That would mean that all tests for operators (window/ join) that instantiate stores would now be a part of `samza-kv-rocksdb`. Let's sync up offline since the scope of that seems to be larger than this PR.

Also, I believe circular dependencies in `testCompile` tasks are fine in gradle.